syntax = "proto3";
package ipc_interface;

// The request message containing the user's name.
message StoreImageRequest {
  string image_name = 1;
  bytes image = 2;
  // Assume the format for now (jpg)
}

// The response contains an acknowledgement
message StoreImageResponse {
  bool success = 1;
  string str_message = 2;
}

// Register template - template id must be unique
message TemplateMatchRequest {
  string big_image_name = 1;
  string template_image_name = 2;
}

// Register template - template id must be unique
message TemplateMatchResponse {
  bool success = 1;
  int32 x_location = 2;
  int32 y_location = 3;
}

